### ISSUE TITLE
pow of 0 with negative infinity should give an error

### ISSUE THREAD

**Issue by Yehouda** [y2q-actionman/with-c-syntax#6]
Line 209 in test/libc_math.lisp:

`is (pow (-0.0, double-float-negative-infinity) == double-float-positive-infinity);`

Calls **(expt -0.0 double-float-negative-infinity)**. According to the ANSI standard:

>  If b=0 and the real part of x is strictly positive, then b^x=0. For all other values of x, 0^x is an error. 

So the line should give an error, and on LispWorks it does. 
http://www.lispworks.com/documentation/HyperSpec/Body/f_exp_e.htm#expt

**Comment by y2q-actionman** [comment]
I'm now confirming it. Sorry for delaying.

**Comment by y2q-actionman** [comment]
I confirmed the Hyperspec. You are correct.

I also checked the specification of `pow()` function in [C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf). The standard says in pp.229:   

> double pow(double x, double y);

> A domain error or range error may occur if x is zero and y is less than zero.

So, as a libc test, I had to consider the possibility of occurring a mathematical error on calling `pow(-0.0, -INFINITY);`.
I fixed it on commit e93709edaf0d08b6cba09be0795be2dc50113d99.

____

By the way, I see Allegro CL 10.1 on MacOSX 11.2.3 raises an error like Lispworks:

> CL-USER(5): (expt -0.0 excl:*negative-infinity-double*)
> Error: attempt to raise 0.0 to non-positive power
>   [condition type: SIMPLE-ERROR]

But, SBCL 2.1.4 does not so:

> * (expt -0.0 sb-ext:double-float-negative-infinity)
> #.DOUBLE-FLOAT-POSITIVE-INFINITY

I'm going to look into SBCL when I have time.
